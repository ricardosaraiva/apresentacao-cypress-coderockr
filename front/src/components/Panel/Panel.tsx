import { FC } from 'react';
import {
  Container,
} from './styles';

const Panel: FC<{}> = ({ children }) => (
  <Container className="panel">
    { children }
  </Container>
);

export default Panel;
