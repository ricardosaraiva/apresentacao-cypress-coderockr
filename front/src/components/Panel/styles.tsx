import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    position: relative;
    z-index: 1;
    color: ${({ theme }) => theme.palette.orange._500.toString()};
    background: ${({ theme }) => theme.palette.white.toString()};
    padding: 30px;
    text-align: center;
    box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.2), 0 5px 5px 0 rgba(0, 0, 0, 0.24);
`;
