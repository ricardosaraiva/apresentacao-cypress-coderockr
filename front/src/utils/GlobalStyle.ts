import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap');

  body, *, input, textarea, select, button {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
    font-size: 16px;
    font-family: 'Roboto', sans-serif !important;
    outline: none !important;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  body {
    background-image: linear-gradient(
      to bottom right,
      ${({ theme }) => theme.palette.orange._500.toString()},
      ${({ theme }) => theme.palette.orange._400.toString()}
    );
    width: 100%;
    height: 100vh;
    color: ${({ theme }) => theme.palette.white.toString()};
  }
`;
