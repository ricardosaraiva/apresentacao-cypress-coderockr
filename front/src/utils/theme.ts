import 'styled-components';
import color from 'color';

const theme = {
  palette: {
    grey: {
      _50: color('#e3e7eb'),
      _100: color('#fafafa'),
      _700: color('#666'),
    },
    black: color('#000'),
    white: color('#FFF'),
    orange: {
      _400: color('#f7c974'),
      _500: color('#F2A91C'),
      _700: color('#af7303'),
    },
    green: {
      _500: color('#246224'),
    },
    red: {
      _500: color('#AA0601'),
    },
  },
};

type Theme = typeof theme

declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}

export default theme;
