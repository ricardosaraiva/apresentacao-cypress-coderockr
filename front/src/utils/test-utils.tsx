import { ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Router } from 'react-router-dom'
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { Model, Registry, Response, Server } from 'miragejs';

import theme from './theme';
import Schema from 'miragejs/orm/schema';
import { ModelDefinition } from 'miragejs/-types';

const history = createMemoryHistory();

type User = {
  id: number,
  name: string,
  email: string,
  password: string,
};

const UserModel: ModelDefinition<User> = Model.extend({});

type AppRegistry = Registry<{ users: typeof UserModel }, { }>
type AppSchema = Schema<AppRegistry>

const makeServer = (): Server => new Server({
  models: {
    users: Model,
  },
  routes: function() {
    this.timing = 100;
    this.urlPrefix = 'http://localhost:8080';
    this.logging = false;

    this.get('/user/me', async (schema: AppSchema, request) => {
      const user = await schema.find('users', request.queryParams.id);

      if (!user) {
        return new Response(400, {}, 'Usuário não encontrado');
      }

      return user.attrs;
    });

    this.post('/user/login', async (schema: AppSchema, request) => {
      const body = JSON.parse(request.requestBody);
      const user = await schema.findBy('users', {
        email: body.email,
        password: body.password,
      });

      if(!user) {
        return new Response(400, {}, 'Usuário não encontrado');
      }

      return user.attrs;
    });

    this.post('/user/create', async (schema: AppSchema, request) => {
      const body = JSON.parse(request.requestBody);

      const user = await schema.findBy('users', {
        email: body.email,
        password: body.password,
      });

      if(user) {
        return new Response(400, {}, 'Usuário duplicado');
      }

      await schema.create('users', {
        _id: 1,
        name: body.name,
        email: body.email,
        password: body.password,
      });

      const userData = await schema.find('users', '1');

      return userData?.attrs;
    });
  },
});

const renderSetup = (
  component: ReactNode,
  options?: Omit<RenderOptions, 'queries'>,
) => {
  return render (
    <ThemeProvider theme={theme}>
      <Router history={history}>
        {component}
      </Router>
    </ThemeProvider>,
    options,
  )
};

export * from '@testing-library/react'
export { render as renderWihtoutTheme};
export { renderSetup as render };
export { history };
export { makeServer };
