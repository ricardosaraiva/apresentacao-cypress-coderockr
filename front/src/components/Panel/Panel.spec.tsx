import { render } from '../../utils/test-utils';

import Panel from './Panel';

describe('Componente Panel', () => {
    test('Valida se o componente panel esta renderizando', () => {
        const { container } = render(<Panel>panel</Panel>);

        const el = container.querySelector('.panel') as Element;

        expect(window.getComputedStyle(el)).toMatchSnapshot();
        expect(container).toMatchSnapshot();
    });
})
