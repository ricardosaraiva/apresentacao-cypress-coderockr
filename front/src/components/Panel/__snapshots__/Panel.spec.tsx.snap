// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Panel Valida se o componente panel esta renderizando 1`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "width",
  "2": "position",
  "3": "z-index",
  "4": "color",
  "5": "background",
  "6": "padding",
  "7": "text-align",
  "8": "box-shadow",
  "9": "visibility",
  "_importants": Object {
    "background": "",
    "box-shadow": "",
    "color": "",
    "display": "",
    "padding": "",
    "position": "",
    "text-align": "",
    "visibility": undefined,
    "width": "",
    "z-index": "",
  },
  "_length": 10,
  "_onChange": [Function],
  "_values": Object {
    "background": "rgb(255, 255, 255)",
    "background-color": "rgb(255, 255, 255)",
    "box-shadow": "0 0 20px 0 rgba(0,0,0,0.2),0 5px 5px 0 rgba(0,0,0,0.24)",
    "color": "rgb(242, 169, 28)",
    "display": "block",
    "padding": "30px",
    "padding-bottom": "30px",
    "padding-left": "30px",
    "padding-right": "30px",
    "padding-top": "30px",
    "position": "relative",
    "text-align": "center",
    "visibility": "visible",
    "width": "100%",
    "z-index": "1",
  },
}
`;

exports[`Componente Panel Valida se o componente panel esta renderizando 2`] = `
<div>
  <div
    class="sc-bdfBwQ bGzPWs panel"
  >
    panel
  </div>
</div>
`;
